File: axhttp/android/retrofit/java-retrofit/demo/java-retrofit-demo/src/main/java/com/axsecurity/sdk/axhttp/retrofit/demo/DemoClientInstance.java
package com.axsecurity.sdk.axhttp.retrofit.demo;

 // *** UNCOMMENT THE LINES BELOW FOR SDK ***
//import com.axsecurity.sdk.axhttp.retrofit.AXHTTPService;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class DemoClientInstance {
    private static final String BASE_URL = "https://example.io";

  // *** COMMENT THE LINES BELOW FOR SDK ***
    private static Retrofit retrofit;
    public static Retrofit getRetrofitInstance() {
        if (retrofit == null) {
            retrofit = new retrofit2.Retrofit.Builder()
                    .baseUrl(BASE_URL)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();
        }
        return retrofit;
    }
    
     // *** UNCOMMENT THE LINES BELOW FOR SDK ***
    // private static Retrofit retrofit;
    // public static Retrofit getRetrofitInstance() {
    //     if (retrofit == null) {
    //         Retrofit.Builder builder = new retrofit2.Retrofit.Builder()
    //                 .baseUrl(BASE_URL)
    //                 .addConverterFactory(GsonConverterFactory.create());
    //         retrofit = AXHTTPService.getRetrofit(builder);
    //     }
    //     return retrofit;
    // }
}

File: axhttp/android/retrofit/kotlin-retrofit/demo/kotlin-retrofit-demo/src/main/java/com/axsecurity/sdk/axhttp/retrofit/demo/DemoClientInstance.kt
package com.axsecurity.sdk.axhttp.retrofit.demo
import com.axsecurity.sdk.axhttp.retrofit.AXHTTPService
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory


object DemoClientInstance {
    private const val BASE_URL = "https://example.io"

    private var retrofit: Retrofit? = null
    @JvmStatic
    val retrofitInstance: Retrofit?
        get() {
            if (retrofit == null) {
                retrofit = Retrofit.Builder()
                    .baseUrl(BASE_URL)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build()
            }
            return retrofit
        }

    // *** UNCOMMENT THE LINES BELOW FOR AXIS ***
//    private var retrofit: Retrofit? = null
//    @JvmStatic
//    val retrofitInstance: Retrofit?
//        get() {
//            if (retrofit == null) {
//                val builder = Retrofit.Builder()
//                        .baseUrl(BASE_URL)
//                        .addConverterFactory(GsonConverterFactory.create())
//                retrofit = AXHTTPService.getRetrofit(builder)
//            }
//            return retrofit
//        }
}